plugins/notifications-backend-module-email-notification-processor/src/module.ts

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  coreServices,
+  createBackendModule,
+} from '@backstage/backend-plugin-api';
+import {
+  NotificationSeverity,
+  Notification,
+} from '@backstage/plugin-notifications-common';
+import {
+  notificationsProcessingExtensionPoint,
+  NotificationProcessor,
+} from '@backstage/plugin-notifications-node';
+import { createTransport } from 'nodemailer';
+
+function severityToNumber(severity: NotificationSeverity | undefined): number {
+  switch (severity) {
+    case undefined:
+      return Number.MIN_VALUE;
+    case 'critical':
+      return 4;
+    case 'high':
+      return 3;
+    case 'normal':
+      return 2;
+    case 'low':
+      return 1;
+    default:
+      throw new Error(`notification severity has an unknown value ${severity}`);
+  }
+}
+
+class EmailNotificationProcessor implements NotificationProcessor {
+  threshold: number;
+  sender: string;
+  recipient: string;
+  transport: any;
+
+  constructor(
+    threshold: number,
+    sender: string,
+    recipient: string,
+    hostname: string,
+    port: number,
+    secure: boolean,
+    requireTLS: boolean,
+    username: string,
+    password: string,
+  ) {
+    this.threshold = threshold;
+    this.sender = sender;
+    this.recipient = recipient;
+    this.transport = createTransport({
+      host: hostname,
+      port: port,
+      secure: secure,
+      requireTLS: requireTLS,
+      auth: {
+        user: username,
+        pass: password,
+      },
+      logger: true,
+    });
+  }
+
+  async send(notification: Notification): Promise<void> {
+    const numericSeverity = severityToNumber(notification.payload.severity);
+
+    if (numericSeverity < this.threshold) {
+      return;
+    }
+
+    await this.transport.sendMail({
+      from: this.sender,
+      to: this.recipient,
+      subject: notification.payload.title,
+      text: JSON.stringify(notification),
+    });
+  }
+}
+
+export const notificationsModuleEmailNotificationProcessor =
+  createBackendModule({
+    pluginId: 'notifications',
+    moduleId: 'email-notification-processor',
+    register(reg) {
+      reg.registerInit({
+        deps: {
+          config: coreServices.rootConfig,
+          notifications: notificationsProcessingExtensionPoint,
+        },
+        async init({ config, notifications }) {
+          const emailConfig = config.getOptionalConfig(
```

Comment:
There's a really nice config system that you should be using, it's well documented here: https://backstage.io/docs/conf/defining

Reply:
i do not want to complicate the plugin code

Reply:
Sorry, this is the common convention and must be used. This helps us avoid undocumented config and helps drive the backstage-cli to validate the schema. Many adopters use that to make sure there are no errors in thier `app-config.yaml` in their CI by not using this it would break for them.

Reply:
i searched for an example. i do not see a real use of this feature. I see a plugin defining it but in code does what I do for reading the config. i looked at linguist-backend

Reply:
Did you read the documentation I shared? It explains it: https://backstage.io/docs/conf/defining. I wasn't saying that everything you've done was not correct but there is more to it. You need to create a `config.d.ts` file and define your schema there and update your `package.json` accordingly.

Reply:
I read the doc. it has almost nothing. this schema is used maybe for validation. I tried configuring invalid properties. 'yarn backstage-cli config:check' does not report a problem

Reply:
added config.d.ts

Reply:
Awesome, thanks 👍